File: ws/sanchez/u2/ws14GUI/StudentAppPY/model/student.py
class Student:
    def __init__(self, id_number="", name="", age=0, career="", grades=None):
        self.id_number = id_number
        self.name = name
        self.age = age
        self.career = career
        self.grades = grades if grades is not None else []

    def get_id_number(self):
        return self.id_number

    def get_name(self):
        return self.name

    def get_age(self):
        return self.age

    def get_career(self):
        return self.career

    def get_grades(self):
        return self.grades

    def calculate_gpa(self):
        if not self.grades:
            return 0
        return sum(self.grades) / len(self.grades)

File: my-fresh-copy/ws/sanchez/u2/ws14GUI/StudentAppJS/model/Student.js
class Student {
    constructor(idNumber, name, age, career, grades) {
        this.idNumber = idNumber;
        this.name = name;
        this.age = age;
        this.career = career;
        this.grades = grades;
    }

    getIdNumber() { return this.idNumber; }
    getName() { return this.name; }
    getAge() { return this.age; }
    getCareer() { return this.career; }
    getGrades() { return this.grades; }

    calculateGPA() {
        if (!this.grades || this.grades.length === 0) return 0;
        const sum = this.grades.reduce((acc, g) => acc + g, 0);
        return sum / this.grades.length;
    }
}